src/main/utils.ts

```diff
@@ -264,3 +267,56 @@ export const installApp = async ({
     });
   }
 };
+
+export async function factoryResetUtility({
+  launcherFileSystem,
+  windowInfoMap,
+  privilegedLauncherWindows,
+  holochainManagers,
+  lairHandle,
+  app,
+}: {
+  launcherFileSystem: LauncherFileSystem;
+  windowInfoMap: WindowInfoRecord;
+  privilegedLauncherWindows?: Record<Screen, BrowserWindow>;
+  holochainManagers: Record<string, HolochainManager>;
+  lairHandle?: ChildProcessWithoutNullStreams;
+  app: Electron.App;
+}) {
+  if (!launcherFileSystem) {
+    throw new Error('LauncherFilesystem is undefined. Aborting Factory Reset.');
+  }
+
+  // 1. Close all windows to prevent chromium related files to be accessed by them
+  Object.values(windowInfoMap).forEach((info) => {
+    info.windowObject.close();
+  });
+  if (privilegedLauncherWindows) {
+    Object.values(privilegedLauncherWindows).forEach((window) => {
+      window.close();
+    });
+  }
+
+  // 2. Stop holochain and lair to prevent files being accessed by them
+  Object.values(holochainManagers).forEach((manager) => {
+    manager.processHandle?.kill();
+  });
+
+  lairHandle?.kill();
+
+  // 3. Remove all data
+  launcherFileSystem.factoryReset();
+
+  // 4. Relaunch
+  const options: Electron.RelaunchOptions = {
+    args: process.argv,
+  };
+  // https://github.com/electron-userland/electron-builder/issues/1727#issuecomment-769896927
+  if (process.env.APPIMAGE) {
+    console.log('process.execPath: ', process.execPath);
+    options.args!.unshift('--appimage-extract-and-run');
+    options.execPath = process.env.APPIMAGE;
+  }
+  app.relaunch(options);
+  app.quit();
```

Comment:
I didn't find a reason why .quit() would not be okay here, so seems like a good change :+1: I hope I'm not missing something that I had in mind when I chose `app.exit()`